R/load_model_out_in_eval_set.R

```diff
@@ -5,16 +5,17 @@
 #' @param target_id The target_id to filter to.
 #' @param eval_set A list specifying the evaluation set, derived from the
 #' eval_sets field of the predeval config.
+#' @param rounds_idx 0-based index of the rounds entry to use.
 #'
 #' @return A data frame containing the model output data.
 #' @noRd
-load_model_out_in_eval_set <- function(hub_path, target_id, eval_set) {
+load_model_out_in_eval_set <- function(hub_path, target_id, eval_set, rounds_idx) {
   conn <- hubData::connect_hub(hub_path)
 
   # filter to the requested target_id
   hub_tasks_config <- hubUtils::read_config(hub_path, config = "tasks")
   round_ids <- hubUtils::get_round_ids(hub_tasks_config)
-  task_groups <- hubUtils::get_round_model_tasks(hub_tasks_config, round_ids[1])
+  task_groups <- hubUtils::get_round_model_tasks(hub_tasks_config, round_ids[rounds_idx + 1])
```

Comment:
```suggestion
  task_groups <- hub_tasks_config$rounds[[rounds_idx + 1L]]$model_tasks
```
I think there is a bug here. `get_round_ids()` with default `flatten = "all"` returns a flattened vector of all round IDs across all `rounds` items. So `round_ids[rounds_idx + 1]` doesn't select a round ID from the correct `rounds` item, it just gets the Nth element of the combined list. This worked in the old code because it only ever expected a single `rounds` item, but will break with multiple.